Scripts/UI/PaletteWindow.cs

```diff
@@ -527,70 +526,89 @@ void DrawOutline(Rect lastRect, int lineThickness, Color color)
-			{
-				tabs.Add(new Tab());
-			}
-			List<Object> selectedObjects = tabs[tabIndex].selectedObjects;
+            while(tabs.Count <= tabIndex)
+            {
+                tabs.Add(new Tab());
+            }
+            List<Object> selectedObjects = tabs[tabIndex].trackedObjects;
             StringBuilder output = new StringBuilder();
             for (int i = 0; i < selectedObjects.Count; i++)
             {
                 if(selectedObjects[i] != null)
                 {
-                    string guid = AssetDatabase.AssetPathToGUID(AssetDatabase.GetAssetPath(selectedObjects[i]));
+                    string assetPath = AssetDatabase.GetAssetPath(selectedObjects[i]);
+                    string guid = AssetDatabase.AssetPathToGUID(assetPath);
 
+                    // Save both the GUID and the path in case the GUID changes
                     output.Append(guid);
+                    output.Append(":");
+                    output.Append(assetPath);
                     output.Append(",");
                 }
             }
             string outputString = output.ToString().TrimEnd(',');
 
-			string key = PlayerPrefKeyPrefix;
-			if(tabIndex > 0)
-			{
-				key += tabIndex;
-			}
-
-			PlayerPrefs.SetString(key, outputString);
+            string key = PlayerPrefKeyPrefix;
+            if(tabIndex > 0)
+            {
+                key += tabIndex;
+            }
+            
+            PlayerPrefs.SetString(key, outputString);
 
-			PlayerPrefs.SetString(key + "-Tab", tabs[tabIndex].Name);
+            PlayerPrefs.SetString(key + "-Tab", tabs[tabIndex].Name);
+            PlayerPrefs.Save();
         }
 
-		void Load(int tabIndex)
+        void Load(int tabIndex)
         {
-			while(tabs.Count <= tabIndex)
-			{
-				tabs.Add(new Tab());
-			}
-
-			tabs[tabIndex].selectedObjects.Clear();
-
-			string key = PlayerPrefKeyPrefix;
-			if(tabIndex > 0)
-			{
-				key += tabIndex;
-			}
-
-			string selectionString = PlayerPrefs.GetString(key);
-            string[] newGUIDs = selectionString.Split(',');
-            for (int i = 0; i < newGUIDs.Length; i++)
+            while(tabs.Count <= tabIndex)
             {
-                string path = AssetDatabase.GUIDToAssetPath(newGUIDs[i]);
+                tabs.Add(new Tab());
+            }
+
+            tabs[tabIndex].trackedObjects.Clear();
+
+            string key = PlayerPrefKeyPrefix;
+            if(tabIndex > 0)
+            {
+                key += tabIndex;
+            }
+
+            string trackedString = PlayerPrefs.GetString(key);
+            string[] trackedObjectStrings = trackedString.Split(',');
+            foreach (string trackedObjectString in trackedObjectStrings)
+            {
+                // Get the guid, for older versions this is the only thing in the string, for newer versions it's the
+                // left side of a colon. Either way Split()[0] works  
+                string guid = trackedObjectString.Split(':')[0];
```

Comment:
warning CS0219: The variable `guid' is assigned but its value is never used

Reply:
Fixed